File: barq-sdk-go/observability_smoke_test.go
package barq

import (
	"context"
	"os"
	"testing"

	pb "github.com/YASSERRMD/barq-db/barq-sdk-go/proto"
)

func TestObservabilityClientReadsMetricsAndClusterStatus(t *testing.T) {
	client := NewClient(Config{
		BaseURL: getenv("BARQ_BASE_URL", "http://127.0.0.1:8080"),
		APIKey:  os.Getenv("BARQ_API_KEY"),
	})

	metrics, err := client.GetMetrics(context.Background())
	if err != nil {
		t.Fatalf("get metrics: %v", err)
	}
	if len(metrics.Definitions) == 0 {
		t.Fatal("expected metrics definitions to be populated")
	}
	if metrics.Storage == nil || metrics.Storage.TotalResidentVectorMemoryBytes == 0 {
		t.Fatal("expected storage metrics to include resident memory")
	}

	status, err := client.GetClusterStatus(context.Background())
	if err != nil {
		t.Fatalf("get cluster status: %v", err)
	}
	if status.Mode != pb.ClusterMode_CLUSTER_MODE_SINGLE_NODE {
		t.Fatalf("unexpected cluster mode %v", status.Mode)
	}
	if status.NodeCount != 1 {
		t.Fatalf("unexpected node count %d", status.NodeCount)
	}
	if status.ShardCount != 1 {
		t.Fatalf("unexpected shard count %d", status.ShardCount)
	}
}

File: barq-sdk-ts/test/observability_smoke.test.js
const test = require("node:test");
const assert = require("node:assert/strict");
const { BarqClient } = require("../dist/index.js");

test("typescript public client reads metrics and cluster status", async () => {
    const client = new BarqClient({
        baseUrl: process.env.BARQ_BASE_URL || "http://127.0.0.1:8080",
        apiKey: process.env.BARQ_API_KEY ?? "",
    });
    const collection = process.env.BARQ_TEST_COLLECTION;

    const metrics = await client.getMetrics();
    assert.ok(metrics.definitions.length > 0);
    assert.ok(Number(metrics.storage.totalResidentVectorMemoryBytes) > 0);
    assert.ok(metrics.storage.collectionWal.some((sample) => sample.collection === collection));

    const status = await client.getClusterStatus();
    assert.equal(status.mode, "single_node");
    assert.equal(Number(status.nodeCount), 1);
    assert.equal(Number(status.shardCount), 1);

    const segmentInfo = await client.getSegmentInfo(collection);
    assert.equal(segmentInfo.collections[0].collection, collection);
});
